File: TabCommon/CommonLibrary/src/main/java/com/nextzy/library/base/view/holder/base/BaseItem.java
package com.nextzy.library.base.view.holder.base;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by「 The Khaeng 」on 08 Oct 2017 :)
 */

public abstract class BaseItem implements Parcelable{
    private int id;
    private int type;

    public BaseItem( int type ){
        this(0,type);
    }

    public BaseItem( int id, int type ){
        this.id = id;
        this.type = type;
    }

    public int getId(){
        return id;
    }

    public BaseItem setId( int id ){
        this.id = id;
        return this;
    }

    public int getType(){
        return type;
    }

    @Override
    public int describeContents(){
        return 0;
    }

    @Override
    public void writeToParcel( Parcel dest, int flags ){
        dest.writeInt( this.id );
        dest.writeInt( this.type );
    }

    public BaseItem(){
    }

    protected BaseItem( Parcel in ){
        this.id = in.readInt();
        this.type = in.readInt();
    }

    public boolean isItemTheSame( Object baseItem){
        return equals( baseItem );
    }

    public boolean isContentTheSame( Object baseItem){
        return equals( baseItem );
    }

}

File: TabCommon/CommonLibrary/src/main/java/com/nextzy/library/base/view/holder/base/BaseItem.kt
package com.nextzy.library.base.view.holder.base

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by「 The Khaeng 」on 26 Sep 2017 :)
 */

abstract class BaseItem(
        val id: Int = 0,
        var type: Int) : Parcelable {

    constructor(source: Parcel) : this(
            source.readInt(), source.readInt() )

    override
    fun describeContents() = 0

    override
    fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(id)
        writeInt(type)
    }
}
